### ISSUE TITLE
TaxHub API

### ISSUE THREAD

**Issue by patkap** [PnX-SI/gn_module_export#28]
>Les différentes instances de TaxHub des structures utilisatrices de GeoNature contiennent aussi des
informations sur les différents taxons (photos, médias, articles liés, descriptions, milieux…).
Le titulaire étudiera dans quelle mesure ces différentes instances pourraient partager des contenus
(https://github.com/PnX-SI/TaxHub/issues/110) mais aussi comment elles pourraient être alimentées
par d’autres sources (INPN, Wikidata, EOL…). En se basant notamment sur le web-sémantique.

**Comment by camillemonchicourt** [comment]
Oui cela est a étudié au niveau de TaxHub où une API assez complète est déjà existante.
Et où des scripts d'intégration de médias provenant de wikidata ou INPN ont été développé entre temps : https://github.com/PnX-SI/TaxHub/issues/150

Reste à voir : 
- pour pouvoir importer des médias depuis l'API GBIF sur le même modèle.
- importer d'autres types de ressources comme des descriptifs ?
- si l'API existante de TaxHub permet de partager des contenus entre différentes instances de TaxHub et comment la sécuriser ou l'ouvrir à tous
- mettre en place le script qui permet d'interroger une API TaxHub pour y récupérer automatiquement les médias et attributs souhaités en pouvant définir des règles (par exemple ne prendre les images que pour les cdnoms où je n'ai aucune image dans mon TaxHub)

**Comment by patkap** [comment]
En ce qui concerne l'enrichissement, @amandine-sahl a tracé une voie des plus élégante et riche.

Quelle solution privilégier pour le partage ?
- une implémentation de système de fichier distribué ([interplanetary filesystem](https://ipfs.io) ou encore [Tahoe-LAFS](https://en.wikipedia.org/wiki/Tahoe-LAFS))
- un module d'import et une route d'export dédiés derrière un algo de diff et synchro avec la duplication inhérente des ressources

**Comment by camillemonchicourt** [comment]
Oui concernant les médias se pose aussi la question de stocker les médias localement ou juste de stocker leur URL distantes.
Les 2 se discutent.
Dans ses scripts INPN et Wikidata @amandine-sahl a plutôt opté pour un import local.

Quant aux 2 solutions sur le partage je ne comprends pas assez pour me prononcer.

**Comment by orovellotti** [comment]
Il y a deux options : 

**1/ Approche base de données distribuée, LOD:**
La phylosophie du Web de données est celle d'une base de données distribuée ou on peut ajouter un nouveau noeud à tout momement en le liant à un noeud exisitant.
Chaque noeuds est ainsi responsable de ses propres données et doit être en ligne pour etre interrogé.

Pour respecter cette vision, il serait intéressant  de ne pas rapatrier les données localements.
Si nous avions une UI pour TaxHub ou pour l'atlas nous pourrions alors lancer la requete sur le LOD a chaque chargement de la page.

**2/Approche base centralisée:**
Nous pourrions aussi rapatrier les informations sur un serveur GeoNature (Cf [Amandine](https://github.com/PnX-SI/TaxHub/issues/150))

Deux problémes ce posent:
a) juqu'ou aller, en effet le Web de donnée est ouvert et donc on ne peut pas tout rapatrier
b) que faire de la responsabilité, si la source change comment le savoir et comment mettre à jour le server local.

Je privilégirais l'option 1 avec une page dynamique (dans l'Atlas) qui permettrais de découvir les données associées présente sur le LOD.

**Comment by patkap** [comment]
Oui, un script est évidement une solution pragmatique à un problème de première nécessité.

Ce que propose Olivier dans son option d'approche "distribuée", dans ma compréhension, serait d'embarquer des données sémantiques dans le code html des pages de taxhub et occtax et de rendre ainsi leur URI déréférençables (_"To dereference a URI means to request a representation of the resource designated by the URI."_).

Ces données pourrait être aussi composées d'URIs déréférençables, nous liant au reste du lod (par l'intermédiaire de taxref-ld et mnhn, gbif, dbpedia, ... par exemple) avec la liberté de mutualiser toutes les ressources.

**Comment by camillemonchicourt** [comment]
Oui OK dans ce cas ça serait au niveau de GeoNature-atlas, car c'est là que les données sont vraiment accessibles, exposées et publiques. 
Mais je pense qu'on verra ça dans un prochain projet.
